signatures/src/main/scala/tsec/signature/imports/JCASigner.scala

```diff
@@ -1,24 +1,46 @@
 package tsec.signature.imports
 
-import cats.instances.either._
-import tsec.signature.core.{SigAlgoTag, SignaturePrograms}
+import cats.effect.Sync
+import tsec.signature.core.{CryptoSignature, SigAlgoTag, SignaturePrograms}
 
-sealed abstract case class JCASigner[A: SigAlgoTag](
-    alg: JCASigInterpreterImpure[A]
-)
-    extends SignaturePrograms[SigErrorM, A] {
+sealed abstract case class JCASigner[F[_]: Sync, A: SigAlgoTag](
+    alg: JCASigInterpreterPure[F, A]
+) extends SignaturePrograms[F, A] {
 
   type PubK  = SigPublicKey[A]
   type PrivK = SigPrivateKey[A]
   type Cert  = SigCertificate[A]
-  val algebra: JCASigInterpreterImpure[A] = alg
+  val algebra: JCASigInterpreterPure[F, A] = alg
```

Comment:
Why is this still `...Pure`?

(Also git's move detection is truly trash here...)

Reply:
Good point. There's still a few `Pure` as far as the interpreters, but those shouldn't be referenced by the user. 

That said, good catch, I'll make it consistent.